allauth/account/views.py

```diff
@@ -290,6 +290,13 @@ def get_context_data(self, **kwargs):
         )
         return ret
 
+    def get_initial(self):
+        if self.request.GET.get("email", None):
+            initial = super().get_initial()
+            initial["email"] = self.request.GET.get("email")
```

Comment:
I think we need to validate (as in https://docs.djangoproject.com/en/4.2/ref/validators/#emailvalidator) the email to avoid prefilling the form with garbage data?

Reply:
Do we also need to respect EMAIL_MAX_LENGTH?

Reply:
No, the prefilled data will be validated for that anyway, but preventing prefilling to appear when you do `email=no-email-just-spam` would be nice.